go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go

```diff
@@ -527,6 +527,9 @@ func (tpb *tablePlanBuilder) generateInsertStatement() *sqlparser.ParsedQuery {
 func (tpb *tablePlanBuilder) generateInsertPart(buf *sqlparser.TrackedBuffer) *sqlparser.ParsedQuery {
 	if tpb.onInsert == insertIgnore {
 		buf.Myprintf("insert ignore into %v(", tpb.name)
+	} else if tpb.onInsert == insertNormal {
+		// the condition (tpb.onInsert == insertNormal) is true when there is no GROUP BY
+		buf.Myprintf("replace into %v(", tpb.name)
```

Comment:
Still WIP, `replace into` is not valid in:

- Materialize, where GROUP BY is found
- Online DDL, where a UNIQUE KEY is added